system-tests/e2e-testsupport/src/test/java/com/hotels/styx/servers/WiremockStyxRequestAdapterTest.java

```diff
@@ -0,0 +1,179 @@
+        assertThat(adapter.containsHeader("Host"), is(true));
+        assertThat(adapter.containsHeader("host"), is(true));
+    }
+
+    @Test(enabled = false)
+    public void adaptsGetAllHeaderKeys() throws Exception {
+        assertEquals(adapter.getAllHeaderKeys(), ImmutableSet.of("Connection", "user-agent", "Content-Type", "Content-Length", "host"));
+    }
+
+    @Test
+    public void adaptsGetUrl() throws Exception {
+        assertThat(adapter.getUrl(), is("/__admin/mappings/new"));
+    }
+
+    @Test
+    public void adaptsGetAbsoluteUrl() throws Exception {
+        assertThat(adapter.getAbsoluteUrl(), is("http://localhost/__admin/mappings/new?msg=6198.1"));
+    }
+
+    @Test
+    public void adaptsGetMethod() throws Exception {
+        assertThat(adapter.getMethod(), is(RequestMethod.POST));
+    }
+
+    @Test
+    public void adaptsGetHeader() throws Exception {
+        assertThat(adapter.getHeader("Content-Length"), is("246"));
+        assertThat(adapter.getHeader("Foo-Bar"), is(nullValue()));
+    }
+
+    @Test(enabled = false)
+    public void adaptsGetHeaders() throws Exception {
+        HttpHeaders headers = adapter.getHeaders();
+
+        assertThat(headers, is(
+                new com.github.tomakehurst.wiremock.http.HttpHeaders(
+                        wireMockHeader("Content-Type", "application/json; charset=UTF-8"),
+                        wireMockHeader("Host", "localhost"),
+                        wireMockHeader("Connection", "Keep-Alive"),
+                        wireMockHeader("User-Agent", "Apache-HttpClient/4.3.5 (java 1.5)"),
+                        wireMockHeader("Content-Length", "208")
+                )));
+    }
+
+    private HttpHeader wireMockHeader(String key, String value) {
+        return new HttpHeader(key, value);
+    }
+
+    @Test
+    public void adaptsGetBodyWhenAbsent() throws Exception {
+        assertThat(adapter.getBody(), is(content.getBytes(UTF_8)));
+    }
+
+    @Test
+    public void adaptsGetBodyWhenPresent() throws Exception {
+        assertThat(adapter.getBody(), is(content.getBytes(UTF_8)));
+    }
+
+    @Test
+    public void adaptsGetBodyAsStringWhenAbsent() throws Exception {
+        assertThat(adapter.getBodyAsString(), is(content));
+    }
+
+    @Test
+    public void adaptsGetBodyAsStringWhenPresent() throws Exception {
+        assertThat(adapter.getBodyAsString(), is(content));
+    }
+
+    @Test
+    public void adaptsContentTypeHeaderWhenPresent() throws Exception {
+        ContentTypeHeader contentType = adapter.contentTypeHeader();
+
+        assertThat(contentType.encodingPart(), is(Optional.of("UTF-8")));
+        assertThat(contentType.mimeTypePart(), is("application/json"));
+    }
+
+    @Test
+    public void adaptsQueryParameter() throws Exception {
+        QueryParameter msg = adapter.queryParameter("msg");
+
+        assertThat(msg.key(), is("msg"));
+        assertThat(msg.values(), is(ImmutableList.of("6198.1")));
```

Comment:
Please use `Matchers.contains` (statically imported, of course) to verify list contents.